crates/db_views/notification/src/impls.rs

```diff
@@ -88,7 +90,14 @@ impl NotificationView {
         .or(comment::post_id.eq(post::id))
         // Filter out the deleted / removed
         .and(not(post::deleted))
-        .and(not(post::removed)),
+        .and(not(post::removed))
+        .or(modlog::target_post_id.eq(post::id.nullable())),
```

Comment:
Why not move this into L88? 

IE `post::id.eq(notification::post_id.or(modlog::target_post_id))`